samples/net/wpan_serial/sample.yaml

```
@@ -8,7 +8,7 @@ common:
     platform_exclude: pinnacle_100_dvk
 tests:
   sample.net.wpan.serial:
-    filter: dt_chosen_enabled("zephyr,ieee802154")
+    filter: dt_chosen_enabled("zephyr,ieee802154") and (not CONFIG_TRUSTED_EXECUTION_NONSECURE or CONFIG_TFM_BOARD!="")
```

Perhaps this would be clearer:
```suggestion
    filter: dt_chosen_enabled("zephyr,ieee802154") and (not CONFIG_TRUSTED_EXECUTION_NONSECURE or CONFIG_BUILD_WITH_TFM)
```

---

I replaced the expression with a single, already existing symbol